tests/tools/conftest.py

```diff
@@ -0,0 +1,88 @@
+import shutil
+from collections.abc import Callable, Generator
+from typing import TYPE_CHECKING
+
+import pytest
+
+from dda.cli.application import Application
+from dda.utils.fs import Path
+
+if TYPE_CHECKING:
+    from dda.tools.git import Git
+
+
+def clear_cached_config(app: Application) -> None:
+    if hasattr(app.config_file, "model"):
+        del app.config_file.model
+
+    if hasattr(app, "config"):
+        del app.config
+
+    if hasattr(app.tools.git, "author_name"):
+        del app.tools.git.author_name
+
+    if hasattr(app.tools.git, "author_email"):
+        del app.tools.git.author_email
+
+
+# Initialize a dummy repo in a temporary directory for the tests to use
+@pytest.fixture
+def temp_repo(app: Application, temp_dir: Path, set_git_author: None) -> Generator[Path, None, None]:  # noqa: ARG001
+    git: Git = app.tools.git
+    repo_path = temp_dir / "dummy-repo"
+    repo_path.mkdir()  # Don't do exist_ok, the directory should not exist
+    with repo_path.as_cwd():
+        git.run(["init", "--initial-branch", "main"])
+    yield repo_path
+    # Cleanup
+
+    shutil.rmtree(repo_path)
```

Comment:
Unnecessary because the parent directory used by the [underlying fixture](https://github.com/DataDog/datadog-agent-dev/blob/1425a34f443f0b468e1739a06fcf97dfbf632594/tests/conftest.py#L55) is already cleaned up.